### DISCUSSION TITLE
listmode farmat and crystal map

### DISCUSSION THREAD

**Discussion by hannah-saber** [UCL/STIR#1063] [General]
Hi,

    Thanks for sharing your working!
    Here, I am confused to reconstruction images from my listmode data by STIR.
    1. listmode data in events should be .safir farmat. I check examples/SAFIR_genericScanner/coinciencesLM.clm.safir by python numpy, I thought it is binary file(uint8). Should I put two crystal IDs per LOR? Or anything else?
    2. it is easy to understand ring/detector/layer in crystal map, but which space coordinates should I choose to compute? Where should I be the origin of the coordinates?
    I cannot find the answers in web, maybe it look like stupid, but as a beginner, it is hard for me :(
    Thanks again! : )

Hannah

**Comment by danieldeidda** [comment]
Hi Hannah,
list-mode reconstruction is scanner dependent, so according to the scanner you are using to acquire your data STIR may not have a model for it. The Safir format only works for that particular scanner. Currently, as far as I know, STIR has formats implemented for PET rootfiles, Safir, Siemens mMR, and GE signa. If you are using a different scanner a model for this needs to be implemented, and would require the vendor to give access to the list-mode format.

**Reply by hannah-saber** [reply]
hi,
thanks for you reply. We get list-mode data form a real PET scanner, so I think STIR can reconstruct it. Cuz I found this project (exmaples/SAFIR_genericScanner) the head file also describe PET scanner, and can reconstruct a point from data. But I am not sure the list mode data format, and I read the documentation, guess it maybe  uint8?but get max 127 and min -127(crystal IDs?or angthing else? I cannot get more information from readme :(   Or maybe I should try transfer list-mode date to PET rootfiles?
Hannah

**Reply by danieldeidda** [reply]
Hi,
Which scanner do you use? every vendor uses a different  format for the list mode so they are the only one to know this information.  The SAFIR generic scanner needs as input a crystal map so you would need to know the position of each crystal. 
However, the first thing you need is to find out if you have access to the list mode format of your PET scanner. Without that I am afraid that reverse-engineering the format may  be very inaccurate but also very time consuming

**Reply by hannah-saber** [reply]
Hi,
thanks for your replay. I can get the crystal map. And after analyze the crystal map and list mode, I guess the list mode data should be uint8, cuz the value can be fit to detector position (0~179) and 16 per group (16 rings), but I stall cannot know how to confirm it is a pair of detectors for a LOR. Secondly, the origin of crystal map should be cylinder center, cuz the range of x,y,z from negative to positive.
I think maybe I can find an example which there is a array description. : )
Hannah

**Comment by KrisThielemans** [comment]
hi, I've converted this to a Discussion. We can always create an issue later if something for code/doc changes comes up.

**Reply by hannah-saber** [reply]
hi, thanks a lot! 

Hannah
